ravenpy/models/importers.py

```diff
@@ -420,18 +379,191 @@ def extract(self) -> GridWeightsCommand:
+                    #                    [lath[ilat+1,ilon+1], lonh[ilat+1,ilon+1]  ],
+                    #                    [lath[ilat,ilon+1]  , lonh[ilat,  ilon+1]  ]]
+
+                    # tmp = self._routing_data_to_geometry(gridcell_edges, epsg=crs_caea)
+                    # tmp.SwapXY()              # switch lat/lon back
+                    # grid_cell_geom_gpd_wkt[ilat][ilon] = tmp
+
+                    # -------------------------
+                    # EPSG:3573   does not need a swap after transform ... and is much faster than transform with EPSG:3035
+                    # -------------------------
+                    #
+                    # Windows            Python 3.8.5 GDAL 3.1.3 --> lat/lon (Ming)
+                    # MacOS 10.15.6      Python 3.8.5 GDAL 3.1.3 --> lat/lon (Julie)
+                    # Graham             Python 3.8.2 GDAL 3.0.4 --> lat/lon (Julie)
+                    # Graham             Python 3.6.3 GDAL 2.2.1 --> lon/lat (Julie)
+                    # Ubuntu 18.04.2 LTS Python 3.6.8 GDAL 2.2.3 --> lon/lat (Etienne)
+                    #
+                    if osgeo_version < "3.0":
+                        gridcell_edges = [
+                            [
+                                lonh[ilat, ilon],
+                                lath[ilat, ilon],
+                            ],  # for some reason need to switch lat/lon that transform works
+                            [lonh[ilat + 1, ilon], lath[ilat + 1, ilon]],
+                            [lonh[ilat + 1, ilon + 1], lath[ilat + 1, ilon + 1]],
+                            [lonh[ilat, ilon + 1], lath[ilat, ilon + 1]],
+                        ]
+                    else:
+                        gridcell_edges = [
+                            [
+                                lath[ilat, ilon],
+                                lonh[ilat, ilon],
+                            ],  # for some reason lat/lon order works
+                            [lath[ilat + 1, ilon], lonh[ilat + 1, ilon]],
+                            [lath[ilat + 1, ilon + 1], lonh[ilat + 1, ilon + 1]],
+                            [lath[ilat, ilon + 1], lonh[ilat, ilon + 1]],
+                        ]
+
+                    tmp = self._shape_to_geometry(
+                        gridcell_edges, epsg=RoutingProductGridWeightImporter.CRS_CAEA
+                    )
+                    grid_cell_geom_gpd_wkt[ilat][ilon] = tmp
+
+        else:
+
+            for ishape in range(self._nlat):
+
+                idx = np.where(self._input_data[self._netcdf_input_field] == ishape)[0]
+                if len(idx) == 0:
+                    # print(
+                    #     "Polygon ID = {} not found in '{}'. Numbering of shapefile attribute '{}' needs to be [0 ... {}-1].".format(
+                    #         ishape, input_file, key_colname_model, nshapes
+                    #     )
+                    # )
+                    raise ValueError("Polygon ID not found.")
```

Comment:
Should we be a bit more talkative here telling the user which ID can't be found where and in which attribute column?

Reply:
Yes absolutely, this is part of the additional error reporting that I want to implement in this PR. But I wanted to open it early, for you to see.